### PR TITLE
MINOR: Add missing unit tests for Mirror Connect

### PR DESCRIPTION
**Pull request by bmaidics**
Some test cases were missing regarding Mirror Connect. Added some to increase test coverage.

### CODE REVIEW

**Review by viktorsomogyi — APPROVED** [review]

**Comment by viktorsomogyi** [comment]
@mimaison would you please review this?

**Comment by viktorsomogyi** [comment]
retest this please

**Review by viktorsomogyi — CHANGES_REQUESTED** [review]
Just noticed that there is an NPE in one of the failed tests (first assumed it's flaky). Please check it @bmaidics and comment if it's related.

**Comment by bmaidics** [comment]
I ran the tests locally and works for me. So probably not related.

File: connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorHeartbeatTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public class MirrorHeartbeatTaskTest {
+
+    @Test
+    public void testPollCreatesRecords() throws InterruptedException {
+        MirrorHeartbeatTask heartbeatTask = new MirrorHeartbeatTask();
+        heartbeatTask.start(TestUtils.makeProps("source.cluster.alias", "testSource",
+                "target.cluster.alias", "testTarget"));
+        List<SourceRecord> records = heartbeatTask.poll();
+        assertEquals(1, records.size());
+        Map sourcePartition = records.iterator().next().sourcePartition();
```
Comment (mimaison) [inline connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorHeartbeatTaskTest.java]: Can we add types for `sourcePartition` -> `Map<String, ?>`

File: connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java [hunk 2]
```diff
@@ -86,4 +96,66 @@ public void testZeroOffsetSync() {
...
+        String sourceClusterName = "cluster1";
+        ReplicationPolicy replicationPolicy = new DefaultReplicationPolicy();
+        MirrorSourceTask mirrorSourceTask = new MirrorSourceTask(consumer, metrics, sourceClusterName,
+                replicationPolicy, 50);
+        List<SourceRecord> sourceRecords = mirrorSourceTask.poll();
+
+        assertEquals(2, sourceRecords.size());
+        for (int i = 0; i < sourceRecords.size(); i++) {
+            SourceRecord sourceRecord = sourceRecords.get(i);
+            ConsumerRecord consumerRecord = consumerRecordsList.get(i);
```
Comment (mimaison) [inline connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java]: `ConsumerRecord<byte[], byte[]> consumerRecord`

File: connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java [hunk 3]
```diff
@@ -86,4 +96,66 @@ public void testZeroOffsetSync() {
...
+            assertEquals(consumerRecord.key(), sourceRecord.key());
+            assertEquals(consumerRecord.value(), sourceRecord.value());
+            // We expect that the topicname will be based on the replication policy currently used
+            assertEquals(replicationPolicy.formatRemoteTopic(sourceClusterName, topicName),
+                    sourceRecord.topic());
+            // We expect that MirrorMaker will keep the same partition assignment
+            assertEquals(consumerRecord.partition(), sourceRecord.kafkaPartition().intValue());
+            // Check header values
+            List<Header> headerList1 = new ArrayList<>();
```
Comment (mimaison) [inline connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java]: Maybe `expectedHeaders` is a better name?

File: connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java [hunk 4]
```diff
@@ -86,4 +96,66 @@ public void testZeroOffsetSync() {
...
+            consumerRecord.headers().forEach(headerList1::add);
+            List<org.apache.kafka.connect.header.Header> headerList2 = new ArrayList<>();
```
Comment (mimaison) [inline connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java]: Maybe `taskHeaders` is a better name?

File: connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java [hunk 5]
```diff
@@ -86,4 +96,66 @@ public void testZeroOffsetSync() {
...
+            sourceRecord.headers().forEach(headerList2::add);
+            compareHeaders(headerList1, headerList2);
+        }
+    }
+
+    private void compareHeaders(List<Header> headers1, List<org.apache.kafka.connect.header.Header> headers2) {
```
Comment (mimaison) [inline connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorSourceTaskTest.java]: Let's rename `headers1` and `headers2` here too

**Review by mimaison — CHANGES_REQUESTED** [review]
Thanks for the PR. It looks good overall. I've just left a few minor suggestions

**Comment by bmaidics** [comment]
Thanks, @mimaison for the review, I addressed all the comments

**Review by mimaison — APPROVED** [review]
Thanks for the PR, LGTM

**Pull request merged**